pyccel/codegen/printing/ccode.py

```diff
@@ -317,6 +323,32 @@ def __init__(self, filename, prefix_module = None):
         self._current_module = None
         self._in_header = False
 
+    def sort_imports(self, imports):
+        """
+        Sort imports to avoid any errors due to bad ordering.
+
+        Sort imports. This is important so that types exist before they are used to create
+        container types. E.g. it is important that complex or inttypes be imported before
+        vec_int or vec_double_complex is declared.
+
+        Parameters
+        ----------
+        imports : list[Import]
+            A list of the imports.
+
+        Returns
+        -------
+        list[Import]
+            A sorted list of the imports.
+        """
+        import_src = [str(i.source) for i in imports]
+        stc_imports = [i for i in import_src if i.startswith('stc/') or i in import_header_guard_prefix]
+        non_stc_imports = [i for i in import_src if i not in stc_imports]
+        stc_imports.sort()
+        non_stc_imports.sort()
```

Comment:
Are you sure about this sorting? What are you sorting by?

This seems dangerous to me. What happens if you have a list of list of a type beginning with Z?

```python
list[list[Z]]
```
`vector_Z` comes after `vector_vector_Z` but it needs to be printed before.

Reply:
I copied this method from your cspan PR, I didn't really go through it but now that mentioned it you are right it might be problematic

Reply:
I think the issue you hinted at will be best handled elsewhere since it is about sorting import targets and not imports themselves. We can add a method to topologically sort the targets of imports based on their dependency graph.

Reply:
Good idea!

Reply:
I just meant `.sort()` but if it's passing then removing it completely is better. I will have to reexamine #2064